ci-operator/step-registry/openshift-observability/enable-cluster-logging/openshift-observability-enable-cluster-logging-commands.sh

```diff
@@ -0,0 +1,605 @@
+  name: minio-console
+  namespace: ${MINIO_NAMESPACE}
+spec:
+  port:
+    targetPort: 9001
+  to:
+    kind: Service
+    name: minio-service-console
+  tls:
+    termination: edge
+    insecureEdgeTerminationPolicy: Redirect
+EOF
+}
+
+function wait_minio_ready(){
+    echo "### wait the mino ready up to 2 minutes"
+
+    minio_ready=false
+    count=1
+    while [[ $count -lt 8 ]]; do
+        if oc -n ${MINIO_NAMESPACE} wait pod --for=condition=Ready  -l app=minio; then
+            minio_ready=true
+        fi
+        if [[ $minio_ready == "true" ]]; then
+            break
+        fi
+        sleep 15s
+        (( count=count + 1 ))
+    done
+
+    if [[ $minio_ready == "false" ]]; then
+       echo "minio is not ready in 2 minutes"
+       oc -n ${MINIO_NAMESPACE} get pods 
+       exit 1
+    fi
+}
+
+function deploy_lokistack()
+{
+    echo "### deploy lokistack logging-loki"
+    prepare_namespace ${LOKI_NAMESPACE}
+
+    echo " Create lokitack secret s3-secret"
+    oc -n ${LOKI_NAMESPACE} create secret generic s3-secret --from-literal=bucketnames="logging-loki" \
+        --from-literal=region="" \
+        --from-literal=endpoint="https://minio-service.${MINIO_NAMESPACE}.svc:9000" \
+        --from-literal=access_key_id="minio" --from-literal=access_key_secret="${minio_secret}"
+    schema_version="v13"
+    if [[ "$lo_csv" =~ loki-operator.v5 ]];then
+        schema_version="v12"
+    fi
+
+cat <<EOF|oc create -f -
+apiVersion: loki.grafana.com/v1
+kind: LokiStack
+metadata:
+  name: ${LOKI_NAME}
+  namespace: ${LOKI_NAMESPACE}
+spec:
+  managementState: Managed
+  size: ${SIZE}
+  storage:
+    schemas:
+      - effectiveDate: '2023-10-15'
+        version: ${schema_version}
+    secret:
+      name: s3-secret
+      type: s3
+  storageClassName: ${storage_class}
+  tenants:
+    mode: openshift-logging
+  rules:
+    enabled: true
+    selector:
+      matchLabels:
+        openshift.io/log-alerting: 'true'
+    namespaceSelector:
+      matchLabels:
+        openshift.io/log-alerting: 'true'
+EOF
+}
+
+function wait_lokistack_ready(){
+    echo "### Wait the lokistack ready up to 5 minutes"
+
+    lokistack_ready=false
+    count=1
+    # Only validate the ingester pod herae
+    while [[ $count -lt 10 ]]; do
+        if oc -n ${LOKI_NAMESPACE} wait pod --for=condition=Ready  -l app.kubernetes.io/instance=${LOKI_NAME} ; then
+            lokistack_ready=true
+	    break
+        fi
+        sleep 30s
+        (( count=count + 1 ))
+    done
+
+    if [[ $lokistack_ready == "false" ]]; then
+       echo "Exit, lokistack is not ready in 5 minutes"
+       oc -n ${LOKI_NAMESPACE} get pod
+       oc -n ${LOKI_NAMESPACE} get lokistack ${LOKI_NAME} -o jsonpath='.status'
+       exit 1
+    fi
+}
+
+function enable_logging_uiplugin(){
+    echo "### Enable_logging_uiplugin"
+    coo=$(oc get csv -o name |grep observability-operator)
+    if [[ $coo == "" ]];then
+        echo "Skip LoggingUI -- no observability-operator"
```

Comment:
I think no need to check it.

Reply:
We need this information to indicate the COO is not ready.

Reply:
OK.